packages/angular/src/schematics/convert-tslint-to-eslint/convert.ts

```diff
@@ -0,0 +1,840 @@
+      {
+        files: ['*.ts'],
+        extends: tsExtends.length > 0 ? tsExtends : undefined,
+        parserOptions: {
+          project: [`${projectRoot}/tsconfig.*?.json`],
+        },
+        plugins: tsPlugins.length > 0 ? tsExtends : undefined,
+        rules: {
+          ...convertedProjectESLintConfig.rules,
+        },
+      },
+    ];
+
+    // No longer relevant/required
+    delete convertedProjectESLintConfig.parser;
+    delete convertedProjectESLintConfig.parserOptions;
+
+    // All applied in the .ts overrides block so should no longer be at the root of the config
+    delete convertedProjectESLintConfig.rules;
+    delete convertedProjectESLintConfig.plugins;
+
+    return chain([
+      ensureESLintPluginsAreInstalled(convertedProject.ensureESLintPlugins),
+      // Create the .eslintrc.json file in the tree using the finalized config
+      updateJsonInTree(
+        join(normalize(projectRoot), '.eslintrc.json'),
+        () => convertedProjectESLintConfig
+      ),
+      // Delete the project's tslint.json, it's no longer needed
+      (host) => host.delete(projectTSLintJsonPath),
+    ]);
+  };
+}
+
+/**
+ * Remove the relative extends to the root TSLint config before converting,
+ * otherwise all the root config will be included inline in the project config.
+ */
+function removeExtendsFromProjectTSLintConfigBeforeConverting(
+  projectRoot: string,
+  projectTSLintJsonPath: string
+): Rule {
+  return updateJsonInTree(projectTSLintJsonPath, (json) => {
+    if (!json.extends) {
+      return json;
+    }
+    const extendsFromRoot = `${offsetFromRoot(projectRoot)}tslint.json`;
+
+    if (Array.isArray(json.extends) && json.extends.length) {
+      json.extends = json.extends.filter(
+        (ext: string) => ext !== extendsFromRoot
+      );
+    }
+    if (typeof json.extends === 'string' && json.extends === extendsFromRoot) {
+      delete json.extends;
+    }
+    return json;
+  });
+}
+
+/**
+ * Templates and source code require different ESLint config (parsers, plugins etc), so it is
+ * critical that we leverage the "overrides" capability in ESLint.
+ *
+ * We therefore need to split out rules which are intended for Angular Templates and apply them
+ * in a dedicated config block which targets HTML files.
+ */
+function separateCodeAndTemplateRules(
+  convertedESLintConfig: ESLintLinter.Config
+) {
+  const codeRules = convertedESLintConfig.rules || {};
+  const templateRules: ESLintLinter.Config['rules'] = {};
+
+  Object.keys(codeRules).forEach((ruleName) => {
+    if (
+      ruleName.startsWith('@angular-eslint/template') ||
+      ruleName.startsWith('@angular-eslint/eslint-plugin-template')
+    ) {
+      templateRules[ruleName] = codeRules[ruleName];
+    }
+  });
+
+  Object.keys(templateRules).forEach((ruleName) => {
+    delete codeRules[ruleName];
+  });
+
+  return {
+    codeRules,
+    templateRules,
+  };
+}
+
+function dedupeEnvAgainstConfigs(
```

Comment:
These `dedupe` methods seem like duplicates of what we did during the `presets` migration. I think moving this generator to `@nrwl/linter` should enable us to share that code.

Reply:
These are actually an improved version I think, but I'll need to go back and double check